File: 5 kyu/Rot13/Rot13.cpp
#include <string>
using namespace std;

string rot13(string msg)
{
  const int ADD = 13;
  string str = "";
  for (char x: msg) {
    if (x >= 'a' && x <= 'z')
      str += (x+ADD-'a')%26+'a';
    else if (x >= 'A' && x <= 'Z')
      str += (x+ADD-'A')%26+'A';
    else
      str += x;
  }
  return str;
}

File: 5 kyu/ROT13/ROT13.js
function rot13(str) {
  const ADD = 13;
  var r = "";
  for (let i = 0; i<str.length; i++) {
    let code = str.charCodeAt(i);
    if (code >= 65 && code < 65+26)
      r += String.fromCharCode((code+ADD-65)%26+65);
    else if (code >= 97 && code < 97+26)
      r += String.fromCharCode((code+ADD-97)%26+97);
    else
      r += str[i];
  }
  return r;
}
